tests/unittests/client/test_runner.py

```diff
@@ -163,6 +163,37 @@ def slow_gather():
     ), "Trials had time to finish because of the slow gather"
 
 
+class CustomExceptionForTest(Exception):
+    pass
+
+
+def test_interrupted_scatter_gather_custom_signal():
```

Comment:
We should also have a test verifying that the previous signal handlers are properly set back on `__exit__`.

Reply:
done

Reply:
Thanks!